### PR TITLE
feat: Add support for `service_ipv4_cidr` for the EKS cluster

### PR DESCRIPTION
**Pull request by nasuku**
# PR o'clock

## Description

This change adds support for service_ipv4_cidr that has been added to terraform-provider-aws last week.
( please see https://github.com/hashicorp/terraform-provider-aws/issues/15505 )

This allows the user to specify the ipv4 CIDR for the k8s services (resolves https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1056 )

### Checklist

- [x] CI tests are passing
- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by nasuku** [comment]
The linter failures (which are happening on top of tree) are fixed in a separate pr #1140 .

**Comment by goodspellar** [comment]
Is there a rough time estimate of when this might get merged? Looks like a rather benign change, but it's blocking our prod EKS deployment :)

**Comment by snvnandan** [comment]
@nasuku we forked you repo to use it. With the new Service IPv4 CIDR, we ran into `--dns-cluster-ip` issue. We were getting 172.20.0.10 as the cluster DNS on the pods, which is not the subnet we are using. This also needs to be looked into on [bootstrap.sh](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh) in amazon-eks-ami. 

As a workaround we specified this `--dns-cluster-ip <ip>` in `bootstap_extra_args` referenced in [local.tf](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf)

**Comment by nasuku** [comment]
@im-nonedone  thanks for mentioning it. Yes. I too was passsing the dns-cluster-ip in bootstrap_extra_args.
I felt that this was more like a bug/enhancement in bootstrap.sh than handling in terraform.
if no fix comes from aws, then may be, a separate PR which accepts dns_cluster_ip as another variable would be a better option.

File: cluster.tf [hunk 1]
```diff
@@ -22,6 +22,10 @@ resource "aws_eks_cluster" "this" {
     public_access_cidrs     = var.cluster_endpoint_public_access_cidrs
   }
 
+  kubernetes_network_config {
+    service_ipv4_cidr = var.service_ipv4_cidr != "" ? var.service_ipv4_cidr : null
```
Comment (barryib) [inline cluster.tf:26]: wouldn't be simpler if we  define `var.service_ipv4_cidr` default to `null` and then just write:

```hcl
kubernetes_network_config {
    service_ipv4_cidr = var.service_ipv4_cidr
}
```

File: variables.tf [hunk 2]
```diff
@@ -367,3 +367,9 @@ variable "fargate_pod_execution_role_name" {
   type        = string
   default     = null
 }
+
+variable "service_ipv4_cidr" {
```
Comment (barryib) [inline variables.tf:371]: Can you please rename this into `cluster_service_ipv4_cidr` for namming consistency.

File: variables.tf [hunk 3]
```diff
@@ -367,3 +367,9 @@ variable "fargate_pod_execution_role_name" {
...
+  description = "service ipv4 cidr for the kubernetes cluster"
+  type        = string
+  default     = ""
```
Comment (barryib) [inline variables.tf:374]: default to `null` as suggested upper ?

**Comment by nasuku** [comment]
thanks for the review @barryib . Made changes per your suggestion and rebased to  master branch.

**Review by barryib — APPROVED** [review]

**Pull request merged**